<?xml version="1.0" encoding="UTF-8"?>
<robot xmlns:xacro="http://wiki.ros.org/xacro" name="$(arg name)">
    <group name="arm_group">
        <joint name="joint1" />
        <joint name="joint2" />
        <joint name="joint3" />
        <joint name="joint4" />
        <joint name="joint5" />
        <joint name="joint6" />
        <joint name="gripper_controller" />
    </group>

    <group name="gripper_group">
        <joint name="gripper_controller" />
    </group>

    <group_state name="init_pose" group="arm_group">
        <joint name="joint1" value="0" />
        <joint name="joint2" value="0" />
        <joint name="joint3" value="0" />
        <joint name="joint4" value="0" />
        <joint name="joint5" value="0" />
        <joint name="joint6" value="0" />
        <joint name="gripper_controller" value="0" />
    </group_state>
    <virtual_joint name="vitual_joint" type="fixed" parent_frame="basic" child_link="base_link" />
<!--     <disable_collisions link1="base_link" link2="link1" reason="Adjacent" />
    <disable_collisions link1="base_link" link2="link4" reason="Never" />
    <disable_collisions link1="link1" link2="link2" reason="Adjacent" />
    <disable_collisions link1="link1" link2="link4" reason="Never" />
    <disable_collisions link1="link2" link2="link3" reason="Adjacent" />
    <disable_collisions link1="link2" link2="link4" reason="Never" />
    <disable_collisions link1="link3" link2="link4" reason="Adjacent" />
    <disable_collisions link1="link4" link2="link5" reason="Adjacent" />
    <disable_collisions link1="link4" link2="link6" reason="Never" />
    <disable_collisions link1="link5" link2="link6" reason="Adjacent" /> -->



    <disable_collisions link1="base_link" link2="gripper_left1" reason="Never"/>
    <disable_collisions link1="base_link" link2="gripper_left2" reason="Never"/>
    <disable_collisions link1="base_link" link2="gripper_left3" reason="Never"/>
    <disable_collisions link1="base_link" link2="gripper_right1" reason="Never"/>
    <disable_collisions link1="base_link" link2="gripper_right2" reason="Never"/>
    <disable_collisions link1="base_link" link2="gripper_right3" reason="Never"/>
    <disable_collisions link1="base_link" link2="link1" reason="Adjacent"/>
    <disable_collisions link1="gripper_base" link2="gripper_left1" reason="Never"/>
    <disable_collisions link1="gripper_base" link2="gripper_left2" reason="Adjacent"/>
    <disable_collisions link1="gripper_base" link2="gripper_left3" reason="Adjacent"/>
    <disable_collisions link1="gripper_base" link2="gripper_right1" reason="Never"/>
    <disable_collisions link1="gripper_base" link2="gripper_right2" reason="Adjacent"/>
    <disable_collisions link1="gripper_base" link2="gripper_right3" reason="Adjacent"/>
    <disable_collisions link1="gripper_base" link2="link4" reason="Never"/>
    <disable_collisions link1="gripper_base" link2="link5" reason="Never"/>
    <disable_collisions link1="gripper_base" link2="link6" reason="Adjacent"/>
    <disable_collisions link1="gripper_left1" link2="gripper_left2" reason="Default"/>
    <disable_collisions link1="gripper_left1" link2="gripper_left3" reason="Adjacent"/>
    <disable_collisions link1="gripper_left1" link2="gripper_right1" reason="Never"/>
    <disable_collisions link1="gripper_left1" link2="gripper_right2" reason="Never"/>
    <disable_collisions link1="gripper_left1" link2="gripper_right3" reason="Never"/>
    <disable_collisions link1="gripper_left1" link2="link1" reason="Never"/>
    <disable_collisions link1="gripper_left1" link2="link2" reason="Never"/>
    <disable_collisions link1="gripper_left1" link2="link3" reason="Never"/>
    <disable_collisions link1="gripper_left1" link2="link4" reason="Never"/>
    <disable_collisions link1="gripper_left1" link2="link5" reason="Never"/>
    <disable_collisions link1="gripper_left1" link2="link6" reason="Never"/>
    <disable_collisions link1="gripper_left2" link2="gripper_left3" reason="Never"/>
    <disable_collisions link1="gripper_left2" link2="gripper_right1" reason="Never"/>
    <disable_collisions link1="gripper_left2" link2="gripper_right2" reason="Never"/>
    <disable_collisions link1="gripper_left2" link2="gripper_right3" reason="Never"/>
    <disable_collisions link1="gripper_left2" link2="link1" reason="Never"/>
    <disable_collisions link1="gripper_left2" link2="link2" reason="Never"/>
    <disable_collisions link1="gripper_left2" link2="link3" reason="Never"/>
    <disable_collisions link1="gripper_left2" link2="link4" reason="Never"/>
    <disable_collisions link1="gripper_left2" link2="link5" reason="Never"/>
    <disable_collisions link1="gripper_left2" link2="link6" reason="Never"/>
    <disable_collisions link1="gripper_left3" link2="gripper_right1" reason="Never"/>
    <disable_collisions link1="gripper_left3" link2="gripper_right2" reason="Never"/>
    <disable_collisions link1="gripper_left3" link2="gripper_right3" reason="Default"/>
    <disable_collisions link1="gripper_left3" link2="link1" reason="Never"/>
    <disable_collisions link1="gripper_left3" link2="link2" reason="Never"/>
    <disable_collisions link1="gripper_left3" link2="link3" reason="Never"/>
    <disable_collisions link1="gripper_left3" link2="link4" reason="Never"/>
    <disable_collisions link1="gripper_left3" link2="link5" reason="Never"/>
    <disable_collisions link1="gripper_left3" link2="link6" reason="Never"/>
    <disable_collisions link1="gripper_right1" link2="gripper_right2" reason="Default"/>
    <disable_collisions link1="gripper_right1" link2="gripper_right3" reason="Adjacent"/>
    <disable_collisions link1="gripper_right1" link2="link1" reason="Never"/>
    <disable_collisions link1="gripper_right1" link2="link2" reason="Never"/>
    <disable_collisions link1="gripper_right1" link2="link3" reason="Never"/>
    <disable_collisions link1="gripper_right1" link2="link4" reason="Never"/>
    <disable_collisions link1="gripper_right1" link2="link5" reason="Never"/>
    <disable_collisions link1="gripper_right1" link2="link6" reason="Never"/>
    <disable_collisions link1="gripper_right2" link2="gripper_right3" reason="Never"/>
    <disable_collisions link1="gripper_right2" link2="link1" reason="Never"/>
    <disable_collisions link1="gripper_right2" link2="link2" reason="Never"/>
    <disable_collisions link1="gripper_right2" link2="link3" reason="Never"/>
    <disable_collisions link1="gripper_right2" link2="link4" reason="Never"/>
    <disable_collisions link1="gripper_right2" link2="link5" reason="Never"/>
    <disable_collisions link1="gripper_right2" link2="link6" reason="Never"/>
    <disable_collisions link1="gripper_right3" link2="link1" reason="Never"/>
    <disable_collisions link1="gripper_right3" link2="link2" reason="Never"/>
    <disable_collisions link1="gripper_right3" link2="link3" reason="Never"/>
    <disable_collisions link1="gripper_right3" link2="link4" reason="Never"/>
    <disable_collisions link1="gripper_right3" link2="link5" reason="Never"/>
    <disable_collisions link1="gripper_right3" link2="link6" reason="Never"/>
    <disable_collisions link1="link1" link2="link2" reason="Adjacent"/>
    <disable_collisions link1="link1" link2="link3" reason="Default"/>
    <disable_collisions link1="link1" link2="link5" reason="Never"/>
    <disable_collisions link1="link2" link2="link3" reason="Adjacent"/>
    <disable_collisions link1="link2" link2="link4" reason="Never"/>
    <disable_collisions link1="link2" link2="link5" reason="Never"/>
    <disable_collisions link1="link3" link2="link4" reason="Adjacent"/>
    <disable_collisions link1="link3" link2="link5" reason="Never"/>
    <disable_collisions link1="link4" link2="link5" reason="Adjacent"/>
    <disable_collisions link1="link5" link2="link6" reason="Adjacent"/>
</robot>
